src/data/int/basic.lean

```diff
@@ -351,6 +351,10 @@ begin
   cases hk; exact ⟨_, hk⟩
 end
 
+lemma eq_of_nat_abs_eq_nonneg {a b : ℤ} (ha : 0 ≤ a) (hb : 0 ≤ b) (h : nat_abs a = nat_abs b) :
```

Comment:
I think I'd call it `eq_of_nat_abs_eq_of_nonneg`; see also `eq_of_associated_of_nonneg`.